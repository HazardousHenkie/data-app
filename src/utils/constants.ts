enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
