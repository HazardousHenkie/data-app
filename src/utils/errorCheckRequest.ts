import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
