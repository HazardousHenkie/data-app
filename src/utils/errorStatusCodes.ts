enum ERROR_STATUS_CODES {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    SERVER_ERROR = 500,
    SERVICE_UNAVAILABLE = 503,
    NOT_FOUND = 404
}

export default ERROR_STATUS_CODES
