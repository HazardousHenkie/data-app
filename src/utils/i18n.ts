import i18n from 'i18next'
import Backend from 'i18next-xhr-backend'
import LanguageDetector from 'i18next-browser-languagedetector'
import { initReactI18next } from 'react-i18next'

export const appLocales = ['en', 'ja', 'nl']

i18n.use(Backend)
    .use(Backend)
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        debug: false,
        lng: 'en',
        fallbackLng: 'en',
        whitelist: appLocales,
        interpolation: {
            escapeValue: false
        },
        react: {
            wait: true
        }
    })

export default i18n
