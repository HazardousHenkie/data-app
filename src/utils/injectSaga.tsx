import * as React from 'react'
import hoistNonReactStatics from 'hoist-non-react-statics'
import { useStore } from 'react-redux'

import { getInjectors } from './sagaInjectors'
import { InjectSagaParams, InjectedStore } from 'types'

export default function hocWithSaga<P>({ key, saga, mode }: InjectSagaParams) {
  function wrap(
    WrappedComponent: React.ComponentType<P>
  ): React.ComponentType<P> {
    class InjectSaga extends React.Component<P> {
      public static WrappedComponent = WrappedComponent
      public injectors: ReturnType<typeof getInjectors>

      public static displayName = `withSaga(${WrappedComponent.displayName ||
        WrappedComponent.name ||
        'Component'})`

      constructor(props: any, context: any) {
        super(props, context)

        this.injectors = getInjectors(context.store)

        this.injectors.injectSaga(key, { saga, mode }, this.props)
      }

      public componentWillUnmount() {
        this.injectors.ejectSaga(key)
      }

      public render() {
        return <WrappedComponent {...this.props} />
      }
    }

    return hoistNonReactStatics(InjectSaga, WrappedComponent) as any
  }
  return wrap
}

const useInjectSaga = ({ key, saga, mode }: InjectSagaParams) => {
  const store = useStore() as InjectedStore
  React.useEffect(() => {
    const injectors = getInjectors(store)
    injectors.injectSaga(key, { saga, mode })

    return () => {
      injectors.ejectSaga(key)
    }
  }, [key, mode, saga, store])
}

export { useInjectSaga }
