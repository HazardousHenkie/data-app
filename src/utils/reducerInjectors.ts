import invariant from 'invariant'
import isEmpty from 'lodash/isEmpty'
import isFunction from 'lodash/isFunction'
import isString from 'lodash/isString'

import checkStore from './checkStore'
import createReducer from '../reducers'
import { InjectedStore } from 'types'
import { Reducer } from 'redux'

export function injectReducerFactory(
  store: InjectedStore,
  isValid: boolean = false
) {
  return function injectReducer(key: string, reducer: Reducer<object>) {
    if (!isValid) {
      checkStore(store)
    }

    invariant(
      isString(key) && !isEmpty(key) && isFunction(reducer),
      '(app/utils...) injectReducer: Expected `reducer` to be a reducer function'
    )

    if (
      Reflect.has(store.injectedReducers, key) &&
      store.injectedReducers[key] === reducer
    ) {
      return
    }

    store.injectedReducers[key] = reducer
    store.replaceReducer(createReducer(store.injectedReducers))
  }
}

export function getInjectors(store: InjectedStore) {
  checkStore(store)

  return {
    injectReducer: injectReducerFactory(store, true)
  }
}
