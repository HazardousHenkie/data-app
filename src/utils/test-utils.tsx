import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
