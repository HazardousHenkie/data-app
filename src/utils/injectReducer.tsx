import * as React from 'react'
import hoistNonReactStatics from 'hoist-non-react-statics'
import { useStore, ReactReduxContext } from 'react-redux'

import { getInjectors } from './reducerInjectors'
import { InjectReducerParams, InjectedStore } from 'types'

export default function hocWithReducer<P>({
  key,
  reducer
}: InjectReducerParams) {
  function wrap(
    WrappedComponent: React.ComponentType<P>
  ): React.ComponentType<P> {
    class ReducerInjector extends React.Component<P> {
      public static contextType = ReactReduxContext
      public static WrappedComponent = WrappedComponent
      public static displayName = `withReducer(${WrappedComponent.displayName ||
        WrappedComponent.name ||
        'Component'})`

      constructor(props: any, context: any) {
        super(props, context)

        getInjectors(context.store).injectReducer(key, reducer)
      }

      public render() {
        return <WrappedComponent {...this.props} />
      }
    }

    return hoistNonReactStatics(ReducerInjector, WrappedComponent) as any
  }
  return wrap
}

const useInjectReducer = ({ key, reducer }: InjectReducerParams) => {
  const store = useStore() as InjectedStore
  React.useEffect(() => {
    getInjectors(store).injectReducer(key, reducer)
  }, [key, reducer, store])
}

export { useInjectReducer }
