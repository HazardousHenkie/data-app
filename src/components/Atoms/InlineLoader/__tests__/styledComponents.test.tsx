import React from 'react'
import { render } from 'utils/test-utils'

import LoaderWrapper from '../styledComponents'

import 'jest-styled-components'

describe('<LoaderWrapper />', () => {
    it('should render and match the snapshot', () => {
        const { container } = render(<LoaderWrapper />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = render(<LoaderWrapper />)

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })
})
