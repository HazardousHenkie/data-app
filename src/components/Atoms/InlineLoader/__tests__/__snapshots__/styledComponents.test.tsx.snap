// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderWrapper /> should render and match the snapshot 1`] = `
.c0 {
  z-index: 1600;
  position: absolute;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
/>
`;
